pkg/terra/client/client.go

```diff
@@ -12,27 +12,43 @@ import (
 	"strconv"
 	"time"
 
-	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
-	"github.com/cosmos/cosmos-sdk/types/tx/signing"
-
-	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+	"github.com/pkg/errors"
 
+	cosmosclient "github.com/cosmos/cosmos-sdk/client"
 	tmtypes "github.com/cosmos/cosmos-sdk/client/grpc/tmservice"
+	"github.com/cosmos/cosmos-sdk/codec/legacy"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/std"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
-	"github.com/smartcontractkit/terra.go/msg"
-	wasmtypes "github.com/terra-money/core/x/wasm/types"
-
-	cosmosclient "github.com/cosmos/cosmos-sdk/client"
-	"github.com/pkg/errors"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
 	"github.com/terra-money/core/app"
+	"github.com/terra-money/core/app/params"
+	wasmtypes "github.com/terra-money/core/x/wasm/types"
 
 	"github.com/smartcontractkit/terra.go/key"
+	"github.com/smartcontractkit/terra.go/msg"
 	"github.com/smartcontractkit/terra.go/tx"
 )
 
+var encodingConfig = params.MakeEncodingConfig()
+
+func init() {
+	// Based on app.MakeEncodingConfig
+	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
```

Comment:
let's add a comment that these functions can panic when called more than once

Reply:
:heavy_check_mark: